### ISSUE TITLE
Unable to set variable of type pipelineReturnValue with azurerm_data_factory_pipeline

### ISSUE THREAD

**Issue by Azilyss** [hashicorp/terraform-provider-azurerm#24612]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.1.9

### AzureRM Provider Version

3.50.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory_pipeline

### Terraform Configuration Files

```hcl
terraform {
  backend "local" {}
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.50.0"
    }
  }
}

provider "azurerm" {
  features {}
  subscription_id = local.subscription_id
}

resource "azurerm_resource_group" "test_resource_group" {
    name     = "test_resource_group"
    location = "westeurope"
}

resource "azurerm_data_factory" "data_factory_test" {
  name                            = "data-factory-test"
  location                        = azurerm_resource_group.test_resource_group.location
  resource_group_name             = azurerm_resource_group.test_resource_group.name
  identity {
    type = "SystemAssigned"
  }
}

resource "azurerm_data_factory_pipeline" "pipeline" {
  lifecycle {
    create_before_destroy = true
  }
  data_factory_id = azurerm_data_factory.data_factory_test.id
  name            = "name"
  folder          = "folder"
  concurrency     = 10
  activities_json = jsonencode([
    {
      "name" : "Set pipeline_return_value variable",
      "type" : "SetVariable",
      "dependsOn" : [],
      "policy" : {
        "secureOutput" : false,
        "secureInput" : false
      },
      "userProperties" : [],
      "typeProperties" : {
        "variableName" : "pipelineReturnValue",
        "value" : [
          {
            "key" : "pipeline_return_value_variable",
            "value" : {
              "type" : "String",
              "content" : "abcd"
            }
          }
        ],
        "setSystemVariable" : true
      }
    }
  ])
}
```

### Debug Output/Panic Output

```shell
See the relevant logs in the following section.
```

### Expected Behaviour

A SetVariable activity is created with a variable of type **pipelineReturnValue**. This variable can be retrieved in a parent pipeline through `@activity('activity-name').output.pipelineReturnValue`.

### Actual Behaviour

Upon `terraform plan` and `terraform apply` for the first time, we get the following logs :

```
+ name           = "Set variable"
+ policy         = {
    + secureInput  = false
    + secureOutput = false
  }
+ type           = "SetVariable"
+ typeProperties = {
    + setSystemVariable = true
    + value             = [
        + {
            + key   = "pipeline_return_value_variable"
            + value = {
                + content = "abcd"
                + type    = "String"
              }
          },
      ]
    + variableName      = "pipelineReturnValue"
  }
+ userProperties = []
```

The SetVariable activity is created but the Settings are empty, and the pipelineReturnValue variable is not being set.

Running `terraform plan` again, the logs show the following behaviour :

```
 name           = "Set variable"
~ typeProperties = {
    + setSystemVariable = true
      # (2 unchanged elements hidden)
  }
  # (4 unchanged elements hidden)
},
```

This suggests that the **setSystemVariable** is not correctly being parsed upon running `terraform apply`.

Note that setting a variable of type pipeline variable works without issue.

### Steps to Reproduce

Run 
* `terraform plan`
* `terraform apply`

This will create a SetVariable activity with no settings.

### Important Factoids

NaN

### References

NaN

**Comment by Bangal22** [comment]
me too

**Comment by dmanero** [comment]
yes, I have the same problem

**Comment by Azilyss** [comment]
This issue has been fixed with azurerm version **3.90** @Bangal22 @dmanero.
